ukat/segment/whole_kidney.py

```diff
@@ -0,0 +1,207 @@
+import csv
+import os
+import nibabel as nib
+import numpy as np
+
+from segment import Tkv
+
+
+class Segmentation(nib.Nifti1Image):
+    """
+    Attributes
+    ----------
+    volumes : dict
+        Total, left and right kidney volumes in milliliters.
+    """
+
+    def __init__(self, pixel_array, affine, post_process=True):
+        """Initialise a whole kidney segmentation class instance.
+
+        Parameters
+        ----------
+        pixel_array : np.ndarray
+            Array containing a T2-weighted FSE image.
+        affine : np.ndarray
+            A matrix giving the relationship between voxel coordinates and
+            world coordinates.
+        post_process : bool, optional
+            Default True
+            Keep only the two largest connected volumes in the mask. Note
+            this may cause issue with subjects that have more or less than
+            two kidneys.
+        """
+
+        super().__init__(pixel_array, affine)
+        self.pixel_array = pixel_array
+        self._nifti = nib.Nifti1Image(self.pixel_array, self.affine)
+        self._seg_obj = Tkv(self._nifti)
+        self._mask = self._seg_obj.get_mask(post_process=post_process)
+        self._kidneys = (self._mask > 0.5) * 1
+        self._kidneys[:self.shape[0]//2] *= 2
+        self.volumes = {'total': self._calculate_volume(self._mask > 0.5),
+                        'left': self._calculate_volume(self._kidneys == 1),
+                        'right': self._calculate_volume(self._kidneys == 2)}
+
+    def get_mask(self, binary=True):
+        """
+        Returns the estimated mask from the provided input data.
+
+        Parameters
+        ----------
+        binary : bool, optional
+            Default True.
+            If True, the mask returned will be an array of bools, where True
+            represents voxels that are renal tissue and False represents
+            voxels that are not renal tissue. If False, the mask returned
+            will be the probability that each voxel is renal tissue.
+
+        Returns
+        -------
+        mask : np.ndarray
+            The estimated mask.
+        """
+        if binary:
+            mask = self._mask > 0.5
```

Comment:
Same question about the criteria

Reply:
I see that 0.5 is required in several steps so ignore my previous comment about the 0.5 probability